### DISCUSSION TITLE
Confirmation of Temporal SendAt and Persistence of jobs

### DISCUSSION THREAD

**Discussion by tunechr** [elsa-workflows/elsa-core#2721] [Q&A]
Hi 

I'm just wanted to confirm the scheduled activities behaviour.

We have created an activity inherited from the StartAt activity.
Our activities can be scheduled days apart. 
e.g. 
Day 1 Send Message 
Day 2 Send Another Message 
Day 5 Send Final Message 

Originally we were using Quartz. I noticed that when the server was restarted(local or cloud) the activities would not run on the schedule we had set. 

I could see that there were bookmarks but they would never be processed. 

**I'm assuming this is because Quartz is using an in-memory store by default?**

I installed Hangfire and used a SQL store and this seems to have resolved the activities restarting after a server restart. 

Again just confirming that this is the behaviour we should be seeing and enabling a persistent job store is the way to go?

Cheers
Chris

**Comment by sfmskywalker** [comment]
Even when using the Quartz in-memory store, the scheduled timers should be re-created by Elsa during application startup (based on the bookmarks).

However, you mentioned that you derived a custom activity so maybe there's some condition where the bookmarks are not picked up for some reason.

If you open a new issue with steps to reproduce this behavior, then we can look into it and possibly provide a fix for 2.5.1.
